backend/routes/seatbid.js

```diff
@@ -0,0 +1,186 @@
+import express from 'express';
+import { PrismaClient } from '@prisma/client';
+import { activeBiddings, bids, emitToFlight, expireBidding } from '../websocket.js';
+import { authenticateUser } from '../auth.js';
+import { io } from '../server.js';
+
+const prisma = new PrismaClient();
+const router = express.Router();
+
+/**
+ * @route POST /start
+ * @desc Start a new bidding for a seat
+ * @access Private
+ */
+router.post('/start', authenticateUser, async (req, res) => {
+    const { passengerId, seatNumber, flightId } = req.body;
+    const biddingId = `${flightId}-${seatNumber}`;
+    const expirationTime = Date.now() + 30 * 60 * 1000; // 30 minutes
+
+    const existingBidding = Array.from(activeBiddings.values()).find(
+        bidding => bidding.passengerId === passengerId && bidding.status === 'ACTIVE'
+    );
+
+    if (existingBidding) {
+        return res.status(400).json({ error: 'You already have an active bidding' });
+    }
+
+    const newBidding = {
+        passengerId,
+        seatNumber,
+        flightId,
+        startTime: Date.now(),
+        expirationTime,
+        status: 'ACTIVE'
+    };
+
+    activeBiddings.set(biddingId, newBidding);
+    bids.set(biddingId, []);
+
+    emitToFlight(req.app.locals.io, flightId, 'new-bid', { 
+        biddingId, 
+        seatNumber, 
+        passengerId,
+        bids: []
+    });
+
+    setTimeout(() => expireBidding(req.app.locals.io, biddingId), expirationTime - Date.now());
+
+    res.json({ success: true, biddingId, ...newBidding, bids: [] });
+});
+
+/**
+ * @route POST /bid
+ * @desc Place a bid on an active bidding
+ * @access Private
+ */
+router.post('/bid', authenticateUser, async (req, res) => {
+    const { biddingId, bidderId, amount, bidderSeatNumber } = req.body;
+    const bidding = activeBiddings.get(biddingId);
+
+    if (!bidding || bidding.status !== 'ACTIVE') {
+        return res.status(400).json({ error: 'Bidding not found or not active' });
+    }
+
+    const bidderActiveBidding = Array.from(activeBiddings.values()).find(
+        b => b.passengerId === bidderId && b.status === 'ACTIVE'
+    );
+
+    if (bidderActiveBidding) {
+        bidderActiveBidding.status = 'CANCELLED';
```

Comment:
So we'll automatically cancel previous biddings while bidding in a new auction. 

I would prefer us to allow users to bid in multiple auctions till they still own their seat. Feel free to do that in follow up PR.

Reply:
added in the next PR